pkg/local_object_storage/shard/get.go

```diff
@@ -140,3 +141,51 @@ func (s *Shard) getBytesWithMetadataLookup(addr oid.Address, skipMeta bool) ([]b
 	}
 	return b, err
 }
+
+// GetStream reads an object from shard as a stream. skipMeta flag allows to fetch object from
+// the blobstor directly.
+//
+// Returns the object header and a reader for the payload.
+// The caller is responsible for closing the returned io.ReadCloser if it is not nil.
+//
+// Returns any error encountered that did not allow to completely read the object part.
+// Returns an error of type apistatus.ObjectNotFound if the requested object is missing in shard.
+// Returns an error of type apistatus.ObjectAlreadyRemoved if the requested object has been marked as removed in shard.
+// Returns the object.ErrObjectIsExpired if the object is present but already expired.
+func (s *Shard) GetStream(addr oid.Address, skipMeta bool) (*objectSDK.Object, io.ReadCloser, error) {
```

Comment:
u add same extension for writecache, and then for shard and engine merged w/ the service changes. I'd make extensions for storage components 1st. This will trigger simple refactoring of the service. Test all `GetStream` methods properly. Then, when storage is rdy, change the service